.github/workflows/subgraph-deploy.yaml

```diff
@@ -0,0 +1,76 @@
+name: 'Build and Deploy Subgraph'
+
+on:
+  workflow_dispatch:
+    inputs:
+      Username:
+        description: 'Username for authentication'
+        required: true
+        type: string
+      Password:
+        description: 'Password for authentication'
+        required: true
+        type: string
+      Basic_Auth:
+        description: 'Basic Authentication'
+        required: true
+        type: string
+      Subgraph_Name:
+        description: 'Name of the subgraph to deploy with the pattern <CUSTOMER_NAME>/<SUBGRAPH_NAME>'
+        required: true
+        type: string
+
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    
+    env:
+      #Endpoints
+      INDEXNODE_ENDPOINT: 'index.hosted.protofire-thegraph.com'
+      QUERY_ENDPOINT: 'query.hosted.protofire-thegraph.com'
+      IPFS_ENDPOINT: 'ipfs.hosted.protofire-thegraph.com'
+
+      #Credentials
+      USERNAME: ${{ github.event.inputs.Username }}
+      PASSWORD: ${{ github.event.inputs.Password }}
+      BASIC_AUTH: ${{ github.event.inputs.Basic_Auth }}
+
+      #Subgraph
+      SUBGRAPH_NAME: ${{ github.event.inputs.Subgraph_Name }}
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v2
+
+      - name: Install graph-cli
+        run: yarn global add @graphprotocol/graph-cli@v0.36.1
+
+      - name: Subgraph - Install dependencies
+        run: yarn install
+
+      - name: Subgraph - Codegen
+        run: yarn codegen
+
+      - name: Subgraph - Build
+        run: yarn build
+      
+      - name: Subgraph - Test
+        run: yarn test
+
+      - name: Subgraph - Graph create
+        run: |
+
+          graph create ${{ env.SUBGRAPH_NAME }} \
+                       --version-label ${{ env.SUBGRAPH_NAME }} \
+                       --headers "{\"Authorization\": \"Basic ${{ env.BASIC_AUTH }}\"}" \
+                       --ipfs "https://${{ env.IPFS_ENDPOINT }}" \
+                       --node "https://${{ env.USERNAME }}:${{ env.PASSWORD }}@${{ env.INDEXNODE_ENDPOINT }}"                     
+
+      - name: Subgraph - Graph deploy
+        run: |
+          graph deploy ${{ env.SUBGRAPH_NAME }} \
+                       --version-label ${{ env.SUBGRAPH_NAME }} \
```

Comment:
--version-label should be an input I guess. Each time we deploy a new version we need to provide this input. The input will look like: v1.0.0